tensorflow/python/ops/math_ops.py

```diff
@@ -99,9 +99,51 @@
 from tensorflow.python.util.tf_export import tf_export
 
 # Aliases for some automatically-generated names.
-linspace = gen_math_ops.lin_space
+#linspace = gen_math_ops.lin_space
 nextafter = gen_math_ops.next_after
 
+def linspace(start_in, stop_in, num, name=None, axis=0):
+  if not isinstance(axis, int):
+    raise TypeError('Axis should be an integer. Received an object {0} of type {1}'.format(axis, type(axis)))
+  with ops.name_scope(name, 'linspace', [start_in, stop_in]):
+    start = ops.convert_to_tensor(start_in, name='start')
+    # stop must be convertible to the same dtype as start
+    stop = ops.convert_to_tensor(stop_in, name='stop', dtype=start.dtype)
+    expanded_start = array_ops.expand_dims(start, axis=axis)
+    num_int = array_ops.convert_to_int_tensor(num, name='num')
+    num = cast(num_int, dtype=start.dtype)
+    shape = array_ops.shape(expanded_start)
+    ndims = shape.get_shape()[0].value
+    axis = array_ops.get_positive_axis(axis, ndims)
+    expanded_stop = array_ops.expand_dims(stop, axis=axis)
+    # to avoid having negative values in the range or zero division
+    # The result is sliced in the end so a correct result is returned for num == 1.
+    range_end = gen_math_ops.maximum(num - 1., 1.)
+    delta = (expanded_stop - expanded_start) / range_end
+    shape_range = range(ndims)
```

Comment:
...which can be done by using `ndims = tf.shape(shape)[0]` above and `tf.range(ndims)` here

Reply:
can you also add a test for unknown ndims? Making start a placeholder should do it.

Reply:
`tf.range(None)` does not seem to be supported, at least on my version of the master, so that's why I did not add it (see my comment above about not supporting fully unknown shapes). Am I missing something here?

Reply:
Edit I see what you mean - to get the runtime value, this way we won't be able to validate the axis though, and it looks like in similar operations - like for example `repeat_with_axis` that is used below - unknown `ndims` don't seem to be supported, so I am not sure how to proceed here.

Reply:
Can't you replace repeat_with_axis with broadcast_to?

Reply:
I definitely can, thanks! I will update the PR as soon as I have a few hours to work on this